nginx/Dockerfile.tmpl

```diff
@@ -36,10 +36,20 @@ RUN apk update && \
     rm -rf /var/cache/apk/* && \
     mkdir -p  /etc/nginx/env /run/nginx /srv/www/html /var/log/nginx /var/cache/nginx && \
     sed -i 's/access_log.*/access_log off;/' /etc/nginx/nginx.conf && \
-    mv /tmp/default.conf /etc/nginx/conf.d/default.conf && \
+    mv /tmp/default.conf /etc/nginx/http.d/default.conf && \
     mv /tmp/env.conf /etc/nginx/env/env.conf && \
     mv /tmp/index.html /srv/www/html/index.html && \
-    echo "include /etc/nginx/env/*.conf;" >> /etc/nginx/nginx.conf
+    echo "include /etc/nginx/env/*.conf;" >> /etc/nginx/nginx.conf && \
+    sed -i "s|^\tmap \$http_upgrade|\t# map \$http_upgrade|" /etc/nginx/nginx.conf && \
+    sed -i "s|^\t\tdefault upgrade;|\t#\tdefault upgrade;|" /etc/nginx/nginx.conf && \
+    sed -i "s|^\t\t'' close;|\t#\t'' close;|" /etc/nginx/nginx.conf && \
+    sed -i "s|^\t}|\t# }|" /etc/nginx/nginx.conf && \
+    sed -i "s|http.d/\*\.conf;|http.d/\*\.conf;\n\t}|" /etc/nginx/nginx.conf && \
+    sed -i "s|http.d/\*\.conf;|http.d/\*\.conf;\n\t\t'' close;|" /etc/nginx/nginx.conf && \
+    sed -i "s|http.d/\*\.conf;|http.d/\*\.conf;\n\t\tdefault upgrade;|" /etc/nginx/nginx.conf && \
+    sed -i "s|http.d/\*\.conf;|http.d/\*\.conf;\n\tmap \$http_upgrade \$connection_upgrade {|" /etc/nginx/nginx.conf && \
+    sed -i "s|http.d/\*\.conf;|http.d/\*\.conf;\n\n\t# Helper variable for proxying websockets.|" /etc/nginx/nginx.conf
```

Comment:
I don't understand these. We can just leave this section out, can't we?

Reply:
Yes I think so, it is pretty much useless inside the http.d as it is defined after.
I'll remove it.

Reply:
Done. Also built a new image and aliased it to 1.20 so it can be used immediately.